c/datatablemodule.cc

```diff
@@ -235,7 +235,7 @@ PyInit__datatable()
     init_jay();
 
     py::Frame::Type::init(m);
-    py::Ftrl::Type::init(m);
+    py::PyFtrl::Type::init(m);
```

Comment:
Because I already have `Ftrl` class defined in `ftrl.h`, so would prefer to have different name here. In the current framework, can I simply move `PyFtrl` out of `py` namespace?

Reply:
Yes, I understand there will be no conflict, what I don't like is using the same name for two different things, even if they're in different namespaces. Seems like a potential source of confusion.

Reply:
This is up to you, but keep in mind that we tend to use this kind of convention in other places, when we need to expose an object to python. Things like `py::base_expr`, `py::Frame`, `py::Column`, `py::Rowindex`, etc. This convention is not followed always, but that's mostly due to historic circumstances: there is an element of evolution to this. Let me explain how it came out to be:

Originally we too started with "Py" prefixes: there were "PyDatatable", "PyColumn", "PyRowindex", etc. But this system was confusing in 3 respects:
- it is exactly the same convention as used by the core python. Thus, there is a small chance of accidentally creating an object with the same name as an existing python object. Or vice versa, a new version of Python may add an object that clashes with our existing object. 
- for a person reading the code it may be difficult to understand which of the objects are from python standard library, and which "belong" to datatable.
- prefixing every name with Py is exactly like creating a "fake" namespace. They do it in C because C lacks "real" namespaces, but why should we fake it in C++?

And if you still feel like there is a potential for confusion here, one way to address it is to avoid using the `namespace` keyword. Thus, you'd be forced to always write explicitly `py::Ftrl` or `dt::Ftrl`, which is unambiguous.

Reply:
Thanks for the detailed explanation. Yes, I actually never use `namespace` keyword, because then there is a high chance to get into a trouble. 

What you say makes perfect sense and I would prefer to follow the convention. Will convert it to`py::Ftrl` and `dt::Ftrl` shortly.